hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminMonitorBase.java

```diff
@@ -151,4 +162,34 @@ public int getPendingNodeCount() {
   public Queue<DatanodeDescriptor> getPendingNodes() {
     return pendingNodes;
   }
+
+  /**
+   * If node "is dead while in Decommission In Progress", it cannot be decommissioned
+   * until it becomes healthy again. If there are more pendingNodes than can be tracked
+   * & some unhealthy tracked nodes, then re-queue the unhealthy tracked nodes
+   * to avoid blocking decommissioning of healthy nodes.
+   *
+   * @param unhealthyDns The unhealthy datanodes which may be re-queued
+   * @param numDecommissioningNodes The total number of nodes being decommissioned
+   * @return List of unhealthy nodes to be re-queued
+   */
+  Stream<DatanodeDescriptor> identifyUnhealthyNodesToRequeue(
```

Comment:
nit: Since we are not only identifying but also returning the unhealthy nodes to re-queue, shall we update this method name to `identifyAndGetUnhealthyNodesToRequeue` or `getUnhealthyNodesToRequeue` ?

Reply:
modified to "getUnhealthyNodesToRequeue"